pkg/util/push/otel.go

```diff
@@ -69,6 +70,23 @@ func OTLPHandler(
 		}
 
 		reader := http.MaxBytesReader(nil, r.Body, int64(maxRecvMsgSize))
+		// Handle compression.
+		switch r.Header.Get("Content-Encoding") {
+		case "gzip":
+			gr, err := gzip.NewReader(reader)
+			if err != nil {
+				return nil, err
+			}
+			// Protect against compression blowing up too.
+			reader = http.MaxBytesReader(nil, gr, int64(maxRecvMsgSize))
+
+		case "":
+			// No compression.
+
+		default:
+			return nil, fmt.Errorf("unsupported compression: %s. Only \"none\" and \"gzip\" supported", r.Header.Get("Content-Encoding"))
```

Comment:
it sounds a bit confusing to me if the error says that the value `"none"` is supported when actually the supported none-value is an empty string / the header being unset.

Reply:
Fair, but the way to not have compression in the collector is to explicitly set it to `"none"`. See diff in `development/tsdb-blocks-storage-s3/config/otel-collector.yaml`.

Since the main users will be sending it to us using the Collector, I think the error message is correct. But happy to change it too.

Reply:
ok, fair enough :+1:

Reply:
I agree with @replay. This error is based on encoding specified in `Content-Encoding`. We should return [HTTP status code 415](https://httpwg.org/specs/rfc7231.html#status.415) if possible, and not refer to config options of the collector here. We don't know *what* client is sending the data, it may not be collector at all.